capacity4more/modules/c4m/features_og/c4m_features_og_manage/c4m_manage_categories.inc

```diff
@@ -0,0 +1,834 @@
+<?php
+/**
+ * @file
+ * Custom manage page callbacks for the taxonomy module.
+ */
+
+/**
+ * Form builder for the taxonomy terms overview.
+ *
+ * Display a tree of all the terms in a vocabulary, with options to edit
+ * each one. The form is made drag and drop by the theme function.
+ *
+ * @ingroup forms
+ * @see taxonomy_overview_terms_submit()
+ * @see theme_taxonomy_overview_terms()
+ */
+function c4m_features_og_manage_categories_overview_form($form, &$form_state) {
+  $group = c4m_og_current_group();
+  $vocabulary = c4m_og_vocab_load_group_vocabulary($group, 'c4m_vocab_category');
+
+  $form['#vocabulary'] = $vocabulary;
+  $form['#group'] = $group;
+  $form['#tree'] = TRUE;
+  $form['#parent_fields'] = FALSE;
+
+  $parameters = drupal_get_query_parameters();
+  // Check for confirmation forms.
+  if (isset($parameters['delete'])) {
+    // Delete confirmation form.
+    return array_merge($form, c4m_features_og_manage_taxonomy_term_confirm_delete($form, $form_state, $parameters['delete']));
+  }
+  if (isset($parameters['reset_alphabetical'])) {
+    // Reset alphabetically confirmation form.
+    return c4m_taxonomy_vocabulary_confirm_reset_alphabetical($form, $form_state, $vocabulary->vid, $parameters['reset_alphabetical']);
+  }
+
+  $delta = 0;
+  $term_deltas = array();
+  $menu_item = menu_get_item(current_path());
+  $tree = $menu_item['path'] == 'manage/categories/types' ? taxonomy_get_tree($vocabulary->vid, 0, 1) : taxonomy_get_tree($vocabulary->vid);
+  $term = current($tree);
+  do {
```

Comment:
Use always `while` loops instead of  `do {} while();`.

```
while($term = next($tree)) {
  …
}
```

Or use foreach:

```
foreach ($tree as $term) {
  …
}
```